src/hooks/useAuthenticated.ts

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+import { USER_SELF } from '@/components/constants/url';
+
+export const DEFAULT_AVATAR = '/Avatar.png';
+
+interface Userdata {
+  userName: string;
+  firstName: string;
+  profilePicture: string;
+}
+
+interface HooksReturnType {
+  userData: Userdata;
+  isLoggedIn: boolean;
+  isLoading: boolean;
+}
+
+const useAuthenticated = (): HooksReturnType => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [userData, setUserData] = useState<Userdata>({
+    userName: '',
+    firstName: '',
+    profilePicture: '',
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true);
+      await fetch(USER_SELF, { credentials: 'include' })
+        .then((response) => {
+          if (!response.ok) {
```

Comment:
why are we checking if the `response.ok` fails in `then` block rather than `catch` ?